src/platforms/web/security.js

```diff
@@ -0,0 +1,57 @@
+/* @flow */
+import Vue from 'core/index'
+import {getTrustedTypes, isTrustedValue} from 'shared/util'
+
+type TrustedTypePolicy = {
+  // value returned is actually an object with toString method returning the wrapped value
+  createHTML: (value: any) => string;
+};
+
+let policy: ?TrustedTypePolicy
+// create policy lazily to simplify testing
+function getOrCreatePolicy() {
+  const tt = getTrustedTypes()
+  if (tt && !policy) {
+    policy = tt.createPolicy(Vue.config.trustedTypesPolicyName, {createHTML: (s) => s});
+  }
+
+  return policy
+}
+
+if (process.env.NODE_ENV !== 'production') {
+  // we need this function to clear the policy in tests
+  Vue.prototype.$clearTrustedTypesPolicy = function() {
+    policy = undefined
+  }
+}
+
+// TODO: remove once https://github.com/WICG/trusted-types/issues/36 is resolved.
+function isApplicationUsingTrustedTypes() {
```

Comment:
Please avoid this `try {} catch` block, it's for example possible that a default policy will prevent the exception from being thrown, and TT are still enforced. Why do we need to know if application is using TT here? It should be ok to remove strinigificaton for trusted values regardless of whether the application has a CSP that enforces TT.

Reply:
It's because of the way I implemented `maybeCreateDangerousSvgHTML`. But it's true I forgot about the default policy preventing my hack. I can check if there is a default policy and if `defPolicy.createHTML` doesn't throw.

Reply:
We actually created a implementation that works without throwing and without ugly hacks :)